From 3d07915a47259c4331e9806c6de5598d374edc74 Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Mon, 11 Jan 2016 16:10:10 +0800
Subject: [PATCH] MLK-12181 misc: at25: Cleaning up missing null-terminate in
 conjunction with strncpy

Replacing strncpy with strlcpy to avoid strings that lacks null terminate.

Signed-off-by: Fugang Duan <B38611@freescale.com>
(cherry picked from commit:4b13e9b7cc3024d5700e400419d7e706375b1aa0)

Conflicts:
	drivers/misc/eeprom/at25.c
---
 drivers/misc/eeprom/at25.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/eeprom/at25.c b/drivers/misc/eeprom/at25.c
index 4f3bca1..8984975 100644
--- a/drivers/misc/eeprom/at25.c
+++ b/drivers/misc/eeprom/at25.c
@@ -309,7 +309,7 @@ static int at25_np_to_chip(struct device *dev,
 	u32 val;
 
 	memset(chip, 0, sizeof(*chip));
-	strncpy(chip->name, np->name, sizeof(chip->name));
+	strlcpy(chip->name, np->name, sizeof(chip->name));
 
 	if (of_property_read_u32(np, "size", &val) == 0 ||
 	    of_property_read_u32(np, "at25,byte-len", &val) == 0) {
-- 
1.8.0

